### PR TITLE
Source separation baseline for improvements

### LINKED ISSUE #26: Create pipeline for training Source Separation
Almost done. Models don't train well as data is not normalized. I additionally debug training process from `demucs` and probably will add some enhancements to our pipeline later.

### LINKED ISSUE #30: Write some evaluation tool that allows to detect hardest cases.
This is idea mostly for the Sound Demixing wrong labels challenge. Knowing which example are hard we can suspect these are labeled wrongly. However, this may be also useful in simple training to see where we are poor. As a starting point try to do [this tutorial](https://pytorch.org/audio/main/tutorials/hybrid_demucs_tutorial.html)

### PR DESCRIPTION
**Pull request by SebChw**
There are still some minor issues with Lightning logging - I don't see loss on tqdm loop. Other than that it seems to have similar training dynamics as training on `demucs` repository. 

closes #26 
closes #30

### CODE REVIEW

File: art/configs/datamodule/sdx_labelnoise.yaml
Comment (kordc) [inline art/configs/datamodule/sdx_labelnoise.yaml:1]: why add an empty file?

File: art/configs/logger/neptune.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 _target_: lightning.pytorch.loggers.NeptuneLogger
 api_key: ${oc.env:NEPTUNE_API_TOKEN} # api key is loaded from environment variable
-project: skdbmk/wake-word-detection
+project: skdbmk/sourceseparation
```
Comment (kordc) [inline art/configs/logger/neptune.yaml]: maybe we can have one big Neptune project, where we can train several tasks at once? It'd just require adding a label to each run

Comment (SebChw) [reply art/configs/logger/neptune.yaml]: to be discussed. For me many projects are better.

File: art/configs/datamodule/sdx_labelnoise.yaml
Comment (SebChw) [reply art/configs/datamodule/sdx_labelnoise.yaml:1]: this should be probably filled

**Comment by SebChw** [comment]
After my latest commit, these are introduced changes/bug fixes:

* You can now use mocked logger with `python -m art.train logger=mock`
* Sanity check dataloader won't spawn process after every epoch. Just set persist_workers=True or n_workers=0 
* Now we upload raw `yaml` file as our configuration not dict.
* If you use sanity_check datamodule, please use neptune_no_check/mock logger then. Epochs are going very fast during sanity checking and big models won't be updated + we introduce a lot of network traffic
* Now training continuation works.
